### PR TITLE
Display "My APR" in the topbar next to "My Assets"

### PR DESCRIPTION
**Pull request by mranilrai**
Screenshot 2022-08-10 at 10 26 37 PM

### CODE REVIEW

File: src/components-v2/navbar/NavbarStats.tsx [hunk 1]
```diff
@@ -165,6 +173,20 @@ export const NavbarStats = observer((): JSX.Element => {
...
           />
         </Grid>
+        <Grid item className={classes.assets}>
+          <Tooltip title="Historic APR" arrow>
+            <IconButton className={classes.historicAPR}>
+              <Typography variant="helperText" display="inline">
+                My APR: &nbsp;
+              </Typography>
+              <CurrencyDisplay
+                displayValue={`$${numberWithCommas(myAPR.toFixed())}`}
```
Comment (axejintao) [inline src/components-v2/navbar/NavbarStats.tsx:183]: this shouldn't be a $ amount it should be a %

File: src/mobx/stores/UserStore.ts [hunk 2]
```diff
@@ -48,6 +48,19 @@ export default class UserStore {
...
   }
 
+  get myAPR(): number {
+    const { vaults } = this.store;
+    const deposites: number[] = [];
+    vaults.vaultOrder.flatMap((vault) => {
+      const depositBalance = this.getBalance(vault.vaultToken);
+      deposites.push(depositBalance.price * depositBalance.balance);
+    });
+    const totalAPR =
```
Comment (axejintao) [inline src/mobx/stores/UserStore.ts:58]: not sure what this maths is. 
for each vault you need to grab the apr the user of the vault is experiencing
weight that by the amount of deposit assets into the pool and add those values in an averaged fasion.

**Review by axejintao — CHANGES_REQUESTED** [review]

File: src/mobx/stores/UserStore.ts (hunk 2 above)
Comment (mrbasado) [reply src/mobx/stores/UserStore.ts:58]: Vault APR * ( "user's balance in vault" / "total user balances in all vaults").

1. calculate "total user balances in all vaults" 
2. loop through all vaults the user has a balance in and SUM the above formula up to get the APR for the user

File: src/mobx/stores/UserStore.ts [hunk 3]
```diff
@@ -48,6 +48,24 @@ export default class UserStore {
     return rewardsValue + walletValue;
   }
 
+  get myAPR(): number {
+    const { vaults } = this.store;
+    let totalUserBalance = 0;
+    let userAPR = 0;
+    vaults.vaultOrder.flatMap((vault) => {
```
Comment (mrbasado) [inline src/mobx/stores/UserStore.ts:55]: consider using reduce() for this and APR calucation below

**Review by mrbasado — CHANGES_REQUESTED** [review]

File: src/mobx/stores/UserStore.ts [hunk 4]
```diff
@@ -50,19 +50,17 @@ export default class UserStore {
...
-    let totalUserBalance = 0;
-    let userAPR = 0;
-    vaults.vaultOrder.flatMap((vault) => {
-      const depositBalance = this.getBalance(vault.vaultToken);
-      totalUserBalance += depositBalance.balance;
-    });
-
-    vaults.vaultOrder.flatMap((vault) => {
+    const totalUserBalance = vaults.vaultOrder.reduce(
+      (totalBalance, vault) => totalBalance + this.getBalance(vault.vaultToken).balance,
```
Comment (axejintao) [inline src/mobx/stores/UserStore.ts:54]: use `value`

File: src/mobx/stores/UserStore.ts [hunk 5]
```diff
@@ -50,19 +50,17 @@ export default class UserStore {
...
+      0,
+    );
+    const userAPR = vaults.vaultOrder.reduce((totalAPR, vault) => {
       const depositBalance = this.getBalance(vault.vaultToken);
       if (depositBalance.balance) {
-        userAPR += vault.apr * (depositBalance.balance / totalUserBalance);
+        return totalAPR + vault.apr * (depositBalance.balance / totalUserBalance);
```
Comment (axejintao) [inline src/mobx/stores/UserStore.ts:60]: use `value` here as well

File: src/mobx/stores/UserStore.ts [hunk 6]
```diff
@@ -50,19 +50,17 @@ export default class UserStore {
...
     const { vaults } = this.store;
-    let totalUserBalance = 0;
-    let userAPR = 0;
-    vaults.vaultOrder.flatMap((vault) => {
-      const depositBalance = this.getBalance(vault.vaultToken);
-      totalUserBalance += depositBalance.balance;
-    });
-
-    vaults.vaultOrder.flatMap((vault) => {
+    const totalUserBalance = vaults.vaultOrder.reduce(
```
Comment (axejintao) [inline src/mobx/stores/UserStore.ts:53]:
```
so if they had $10 earning 10% and $20 earning 20%
then you have $30 earning 16.66%
(10 * 10 + 20 * 20) / (10 + 20)
```

processing via this fn

```
totalUserBalance = 10 + 20 = 30

// totalAPR + vault.apr * (depositBalance.balance / totalUserBalance);
useApr = 
0 + 10 * (10 / 30) = 0 + 10 * 0.333 = 3.333
3.333 + 20 * (20 / 30) = 3.333 + 20 * 0.666 = 16.6666
```

code lgtm, just needs to use value

**Review by axejintao — CHANGES_REQUESTED** [review]

**Review by Tritium-VLK — APPROVED** [review]

**Review by mrbasado — APPROVED** [review]

**Review by axejintao — APPROVED** [review]

**Pull request merged**